packages/flame/lib/src/noise.dart

```diff
@@ -0,0 +1,231 @@
+// This code is translated from
+// https://github.com/stegu/perlin-noise/blob/master/src/noise1234.c
+// https://github.com/stegu/perlin-noise/blob/master/src/simplexnoise1234.c
+// by Stefan Gustavson, (c) 2003-2005, released into public domain.
+import 'dart:typed_data';
+
+//#region Perlin noise
+// This implementation is "Improved Noise" as presented by Ken Perlin at
+// Siggraph 2002. The 3D function is a direct port of his Java reference code
+// which was once publicly available on www.noisemachine.com, but the 1D, 2D
+// and 4D functions were implemented by Stefan Gustavson.
+
+/// 1D float Perlin noise
+///
+/// Perlin noise is a procedurally-generated smooth yet chaotic-looking
+/// function that has variety of applications. The function is NOT random: it
+/// returns fixed values for each argument `x`. The return value is in the
+/// range of approximately [-0.752 .. +0.705].
+///
+/// Note that the function returns 0 for every integer value of x.
+double noise1(double x) {
+  var ix0 = x.floor(); // integer part of x
+  final fx0 = x - ix0; // fractional part of x
+  final fx1 = fx0 - 1.0;
+  final ix1 = (ix0 + 1) & 255;
+  ix0 &= 255; // wrap to 0..255
+  final s = _fade(fx0);
+  final n0 = _grad1(_perm[ix0], fx0);
+  final n1 = _grad1(_perm[ix1], fx1);
+  return 0.188 * _lerp(s, n0, n1);
+}
+
+/// 2D float Perlin noise
+double noise2(double x, double y) {
+  var ix0 = x.floor(); // integer part of x
+  var iy0 = y.floor(); // integer part of y
+  final fx0 = x - ix0; // fractional part of x
+  final fy0 = y - iy0; // fractional part of y
+  final fx1 = fx0 - 1;
+  final fy1 = fy0 - 1;
+  final ix1 = (ix0 + 1) & 255; // wrap to 0..255
+  final iy1 = (iy0 + 1) & 255;
+  ix0 &= 255;
+  iy0 &= 255;
+  final t = _fade(fy0);
+  final s = _fade(fx0);
+  final nx00 = _grad2(_perm[ix0 + _perm[iy0]], fx0, fy0);
+  final nx01 = _grad2(_perm[ix0 + _perm[iy1]], fx0, fy1);
+  final n0 = _lerp(t, nx00, nx01);
+  final nx10 = _grad2(_perm[ix1 + _perm[iy0]], fx1, fy0);
+  final nx11 = _grad2(_perm[ix1 + _perm[iy1]], fx1, fy1);
+  final n1 = _lerp(t, nx10, nx11);
+  return 0.507 * _lerp(s, n0, n1);
+}
+
+//#endregion
+
+//#region Simplex noise
+// This is a clean, fast, modern and free Perlin Simplex noise function.
```

Comment:
Isn't this dartdoc?

Reply:
I don't know, this is a comment that relates to all functions in that section, and it was copied from the original Gustavson's implementation

Reply:
Pretty funny that it says modern since the comment was written like 20 years ago 😄